### DISCUSSION TITLE
Tarql?

### DISCUSSION THREAD

**Discussion by bolerio** [linkedpipes/etl#929] [General]
Hi team,

Just discovered this project and excited to try it out. Is there a component that supports Tarql as I have already some Tarql based scripts that I'd prefer to reuse with LinkedPipes. If not, is it possible to write a custom components and any pointers on how to do it? 

Best,
Boris

**Comment by jakubklimek** [comment]
@bolerio There is a [Tabular](https://etl.linkedpipes.com/components/t-tabular) component, converting CSV files to RDF according to the CSV on the Web W3C Recommendation. Then, you can attach a SPARQL Construct component to transform the data coming from Tabular to your desired shape.

As to reusing Tarql scripts, you could probably just add the necessary graph pattern to your WHERE clause matching the Tabular output to your used Tarql variables, given that you do not use some of the Tarql specialities such as `?ROWNUM` - then the adjustments would need to be greater.

As to the custom components, I recommend taking a look at the implementation of some of the [existing ones ](https://github.com/linkedpipes/etl/tree/develop/plugins) and then copy, rename and adjust one.

**Reply by bolerio** [reply]
Thanks @jakubklimek , I will give that a try.
